python/pyspark/mllib/evaluation.py

```diff
@@ -67,6 +67,73 @@ def unpersist(self):
         self.call("unpersist")
 
 
+class RegressionMetrics(JavaModelWrapper):
+    """
+    Evaluator for regression.
+
+    >>> predictionAndObservations = sc.parallelize([
+    ...     (2.5, 3.0), (0.0, -0.5), (2.0, 2.0), (8.0, 7.0)])
+    >>> metrics = RegressionMetrics(predictionAndObservations)
+    >>> metrics.explainedVariance()
+    0.95...
+    >>> metrics.meanAbsoluteError()
+    0.5...
+    >>> metrics.meanSquaredError()
+    0.37...
+    >>> metrics.rootMeanSquaredError()
+    0.61...
+    >>> metrics.r2()
+    0.94...
+    """
+
+    def __init__(self, predictionAndObservations):
+        """
+        :param predictionAndObservations: an RDD of (prediction, observation) pairs.
+        """
+        sc = predictionAndObservations.ctx
+        sql_ctx = SQLContext(sc)
+        df = sql_ctx.createDataFrame(predictionAndObservations, schema=StructType([
+            StructField("prediction", DoubleType(), nullable=False),
+            StructField("observation", DoubleType(), nullable=False)]))
+        java_class = sc._jvm.org.apache.spark.mllib.evaluation.RegressionMetrics
+        java_model = java_class(df._jdf)
+        super(RegressionMetrics, self).__init__(java_model)
+
+    def explainedVariance(self):
```

Comment:
Do you think it is better if we mark it as `@property`? I didn't do that for `BinaryClassificationMetrics`, but we still have time to make this change.